null_safety_examples/misc/lib/articles/creating-streams/stream_controller.dart

```diff
@@ -84,7 +82,7 @@ void showBasicUsage() {
 void demoPause() {
   var counterStream =
       Stream<int>.periodic(Duration(seconds: 1), (x) => x).take(15);
-  StreamSubscription<int> subscription;
+  late StreamSubscription<int> subscription;
```

Comment:
Is there any reason not to merge the declaration and initialisation of this variable? It'd allow skipping the late declaration

Reply:
These were the usages I wasn't completely sure about but to get the original samples working I found the early declaration with `late` is needed as the `subscription` variable is used in the callback. Do you think there is perhaps a different structure that would work better?

Reply:
Actually, if it is being intialised in a callback, that's pretty much `late` by definition. SGTM